internal/handlers/publishers.go

```diff
@@ -58,52 +61,77 @@ func (p *Publisher) GetPublisher(ctx *fiber.Ctx) error {
 
 // PostPublisher creates a new publisher.
 func (p *Publisher) PostPublisher(ctx *fiber.Ctx) error {
-	publisher := new(requests.Publisher)
+	request := requests.Publisher{}
+
+	if err := ctx.BodyParser(&request); err != nil {
+		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json", err)
+	}
 
-	if err := ctx.BodyParser(&publisher); err != nil {
-		return common.Error(fiber.StatusBadRequest, "can't create Publisher", "invalid json")
+	if err := common.ValidateStruct(&request); err != nil {
+		return common.Error(fiber.StatusUnprocessableEntity, "can't create Publisher", "invalid json", err)
 	}
 
-	if err := common.ValidateStruct(*publisher); err != nil {
-		return common.Error(fiber.StatusUnprocessableEntity, "can't create Publisher", "invalid format")
+	publisher := &models.Publisher{
+		ID: utils.UUID(),
+		URLAddresses: []models.URLAddresses{
+			{URL: request.URL},
+		},
 	}
 
 	if err := p.db.Create(&publisher).Error; err != nil {
-		return common.Error(fiber.StatusInternalServerError, "can't create Publisher", "db error")
+		return common.Error(fiber.StatusInternalServerError, "can't create Publisher", "db error", err)
```

Comment:
If we fail here we know it's something unrecoverable related to the database, but I'd leave out the details just in case because the error might leak something about the infrastructure stuff (fe the db hostname?). WDYT?

Reply:
You're right